### PR TITLE
Use dialpad to send dtmf tone 

### PR DESCRIPTION
**Pull request by carocao-msft**
# What
Use dialpad to send dtmf tone 

# Why
https://skype.visualstudio.com/SPOOL/_workitems/edit/2919258

# How Tested
uncomment the button code in call controls and callwithchat controls
join PSTN call with computer and your own phone and open dialpad to test if you can hear dtmf tone 
Note this only works for 1:1 call
Screen Shot 2022-08-10 at 1 31 31 PM

### CODE REVIEW

File: packages/react-composites/src/composites/CallComposite/components/CallControls.tsx [hunk 1]
```diff
@@ -115,6 +147,10 @@ export const CallControls = (props: CallControlsProps & ContainerRectProps): JSX
           )}
           {/* @conditional-compile-remove(control-bar-button-injection) */ customButtons['primary']}
           {isEnabled(options?.endCallButton) && <EndCall displayType={options?.displayType} />}
+          {
+            // uncomment this to test dialpad dtmf
```
Comment (anjulgarg) [inline packages/react-composites/src/composites/CallComposite/components/CallControls.tsx:151]: This comment shouldn't be committed

File: packages/react-components/src/components/Dialpad/Dialpad.tsx [hunk 2]
```diff
@@ -107,6 +107,8 @@ export interface DialpadProps {
   onDisplayDialpadInput?: (input: string) => string;
   /**  on change function for text field */
   onChange?: (input: string) => void;
+  /**  boolean input to determine show/hide delete button, default true */
```
Comment (anjulgarg) [inline packages/react-components/src/components/Dialpad/Dialpad.tsx:110]: determine when* to show/hide

File: packages/react-components/src/components/Dialpad/Dialpad.tsx [hunk 3]
```diff
@@ -183,12 +185,14 @@ const DialpadContainer = (props: {
...
```
Comment (anjulgarg) [inline packages/react-components/src/components/Dialpad/Dialpad.tsx:188]: determine when* to show/hide

File: packages/react-components/src/components/Dialpad/Dialpad.tsx [hunk 4]
```diff
@@ -183,12 +185,14 @@ const DialpadContainer = (props: {
...
+  showDeleteButton?: boolean;
   styles?: DialpadStyles;
 }): JSX.Element => {
   const theme = useTheme();
   const [textValue, setTextValue] = useState('');
 
-  const { onSendDtmfTone, onClickDialpadButton, onDisplayDialpadInput, onChange } = props;
+  const { onSendDtmfTone, onClickDialpadButton, onDisplayDialpadInput, onChange, showDeleteButton } = props;
```
Comment (anjulgarg) [inline packages/react-components/src/components/Dialpad/Dialpad.tsx:195]: Did you forget to assign a default value of `true` to `showDeleteButton` here?

File: packages/react-composites/review/beta/react-composites.api.md [hunk 5]
```diff
@@ -422,6 +425,8 @@ export interface CallWithChatAdapterManagement {
     removeParticipant(userId: string): Promise<void>;
     // @beta
     resumeCall: () => Promise<void>;
+    // @beta
+    sendDtmf: (dtmfTone: DtmfTone) => Promise<void>;
```
Comment (anjulgarg) [inline packages/react-composites/review/beta/react-composites.api.md:429]: sendDtmf -> sendDtmfTone for consistency

File: packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts [hunk 6]
```diff
@@ -288,6 +288,13 @@ export interface CallAdapterCallManagement {
    * @beta
    */
   addParticipant(participant: CommunicationIdentifier, options?: AddPhoneNumberOptions): Promise<void>;
+  /* @conditional-compile-remove(PSTN-calls) */
+  /**
+   * send dtmf tone to another participant in 1:1 call
```
Comment (anjulgarg) [inline packages/react-composites/src/composites/CallComposite/adapter/CallAdapter.ts:293]: in a 1:1 PSTN call*

File: packages/react-composites/src/composites/CallComposite/components/CallControls.tsx [hunk 7]
```diff
@@ -2,6 +2,10 @@
 // Licensed under the MIT license.
 
 import { memoizeFunction, Stack, useTheme } from '@fluentui/react';
+/* @conditional-compile-remove(PSTN-calls) */
+import { useState } from 'react';
+// uncomment this to test dialpad dtmf
+// import { DefaultButton } from '@fluentui/react';
```
Comment (anjulgarg) [inline packages/react-composites/src/composites/CallComposite/components/CallControls.tsx:8]: This should not be in this PR

File: packages/react-composites/src/composites/CallComposite/components/CallControls.tsx [hunk 8]
```diff
@@ -74,8 +92,22 @@ export const CallControls = (props: CallControlsProps & ContainerRectProps): JSX
...
+  const onDismissDialpad = (): void => {
+    setShowDialpad(false);
+  };
+
   return (
     <Stack horizontalAlign="center">
+      {
+        /* @conditional-compile-remove(PSTN-calls) */
+        <SendDtmfDialpad
+          isMobile={props.isMobile ?? false}
```
Comment (anjulgarg) [inline packages/react-composites/src/composites/CallComposite/components/CallControls.tsx:105]: props.isMobile ?? false -> !!props.isMobile

File: packages/react-composites/src/composites/common/SendDtmfDialpad.tsx [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+            <IconButton
+              iconProps={{ iconName: 'Cancel' }}
+              ariaLabel={strings.dialpadCloseModalButtonAriaLabel}
+              onClick={onDismissTriggered}
+              style={{ color: 'black' }}
+            />
+          </Stack>
+
+          <Stack>
+            {' '}
```
Comment (anjulgarg) [inline packages/react-composites/src/composites/common/SendDtmfDialpad.tsx:84]: What's the use of this blank space?

File: packages/react-composites/src/composites/common/SendDtmfDialpad.tsx [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+          onDismiss={onDismissTriggered}
+          isBlocking={true}
+          styles={dialpadModelStyle}
+        >
+          <Stack horizontal horizontalAlign="end" verticalAlign="center">
+            <IconButton
+              iconProps={{ iconName: 'Cancel' }}
+              ariaLabel={strings.dialpadCloseModalButtonAriaLabel}
+              onClick={onDismissTriggered}
+              style={{ color: 'black' }}
```
Comment (anjulgarg) [inline packages/react-composites/src/composites/common/SendDtmfDialpad.tsx:79]: Is the cancel button visible in dark mode?

File: packages/react-composites/src/composites/common/SendDtmfDialpad.tsx [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+    placeholderText: ''
+  };
+
+  if (isMobile) {
+    return (
+      <Stack>
+        {showDialpad && (
+          <Stack styles={drawerContainerStyles}>
+            <_DrawerSurface onLightDismiss={onDismissTriggered}>
+              <Stack style={{ padding: '16px' }}>
```
Comment (anjulgarg) [inline packages/react-composites/src/composites/common/SendDtmfDialpad.tsx:53]: Can we use rem here?

File: packages/react-composites/src/composites/common/SendDtmfDialpad.tsx [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+                {' '}
```
Comment (anjulgarg) [inline packages/react-composites/src/composites/common/SendDtmfDialpad.tsx:54]: What's the use of this blank space?

File: packages/react-composites/src/composites/common/SendDtmfDialpad.tsx [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+
+  const { strings, isMobile, showDialpad, onDismissDialpad } = props;
+
+  const theme = useTheme();
+
+  const onDismissTriggered = (): void => {
+    onDismissDialpad();
+  };
+
+  const dialpadModelStyle: Partial<IModalStyles> = useMemo(() => themedDialpadModelStyle(theme), [theme]);
```
Comment (anjulgarg) [inline packages/react-composites/src/composites/common/SendDtmfDialpad.tsx:39]: model -> modal

File: packages/react-components/src/components/Dialpad/Dialpad.tsx (hunk 4 above)
Comment (carocao-msft) [reply packages/react-components/src/components/Dialpad/Dialpad.tsx:195]: Added default value down below in export const Dialpad = (props: DialpadProps): JSX.Element => {

File: packages/react-composites/src/composites/CallComposite/components/CallControls.tsx (hunk 7 above)
Comment (carocao-msft) [reply packages/react-composites/src/composites/CallComposite/components/CallControls.tsx:8]: This is for the dummy button to activate dialpad. need to uncomment those lines in order to test it. Comment them out so we are not introducing UI changes

File: packages/react-composites/src/composites/common/SendDtmfDialpad.tsx (hunk 9 above)
Comment (carocao-msft) [reply packages/react-composites/src/composites/common/SendDtmfDialpad.tsx:84]: oops typo here

File: packages/react-composites/src/composites/CallComposite/components/CallControls.tsx (hunk 7 above)
Comment (carocao-msft) [reply packages/react-composites/src/composites/CallComposite/components/CallControls.tsx:8]: Updated PR to remove all code for the dummy button

File: packages/react-components/src/components/Dialpad/Dialpad.tsx [hunk 14]
```diff
@@ -183,12 +185,14 @@ const DialpadContainer = (props: {
...
   onChange?: (input: string) => void;
+  /**  boolean input to determine when to show/hide delete button, default true */
+  showDeleteButton?: boolean;
   styles?: DialpadStyles;
 }): JSX.Element => {
   const theme = useTheme();
   const [textValue, setTextValue] = useState('');
 
-  const { onSendDtmfTone, onClickDialpadButton, onDisplayDialpadInput, onChange } = props;
+  const { onSendDtmfTone, onClickDialpadButton, onDisplayDialpadInput, onChange, showDeleteButton } = props;
```
Comment (anjulgarg) [inline packages/react-components/src/components/Dialpad/Dialpad.tsx:195]: Where are we setting the default value true for `showDeleteButton`?

**Review by anjulgarg — APPROVED** [review]

**Review by dmceachernmsft — APPROVED** [review]

**Pull request merged**